<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Example Use Case to counter GMQL • curatedTCGAManu</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Example Use Case to counter GMQL">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">curatedTCGAManu</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/Figure2_OncoPrint.html">PanCan OncoPrint</a>
    </li>
    <li>
      <a href="../../articles/Figures3n4_GSEA_PanCan.html">PanCan DE &amp; GSEA</a>
    </li>
    <li>
      <a href="../../articles/Figures5-6-S4.html">Example Multi-omic Plots</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Example Use Case to counter GMQL</h1>
                        <h4 class="author">Waldron Lab</h4>
            
            <h4 class="date">July 11, 2019</h4>
      
      
      <div class="hidden name"><code>GMQL_vignette.Rmd</code></div>

    </div>

    
    
<p>Load packages:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(MultiAssayExperiment)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(curatedTCGAData)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(TCGAutils)</span></code></pre></div>
<p>From <a href="https://doi.org/10.1093/bioinformatics/bty688">Masseroli et al. 2018</a></p>
<blockquote>
<p>“In TCGA data of BRCA patients, find the DNA somatic mutations within the first 2000 bp outside of the genes that are both expressed with FPKM &gt; 3 and have at least a methylation in the same patient biospecimen, and extract these mutations of the top 5% patients with the highest number of such mutations.”</p>
</blockquote>
<p>First we load the ‘BRCA’ dataset with ‘Mutation’, ‘RNASeq2GeneNorm’ and ‘Methylation’ assays / experiments:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="kw"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(</span>
<span id="cb2-2"><a href="#cb2-2"></a>    brca &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/curatedTCGAData/man/curatedTCGAData.html">curatedTCGAData</a></span>(</span>
<span id="cb2-3"><a href="#cb2-3"></a>        <span class="dt">diseaseCode =</span> <span class="st">"BRCA"</span>,</span>
<span id="cb2-4"><a href="#cb2-4"></a>        <span class="dt">assays =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Mutation"</span>, <span class="st">"RNASeq2GeneNorm"</span>, <span class="st">"Methylation"</span>),</span>
<span id="cb2-5"><a href="#cb2-5"></a>        <span class="dt">dry.run =</span> <span class="ot">FALSE</span></span>
<span id="cb2-6"><a href="#cb2-6"></a>    )</span>
<span id="cb2-7"><a href="#cb2-7"></a>)</span></code></pre></div>
<pre><code>## snapshotDate(): 2020-03-09</code></pre>
<pre><code>## Working on: BRCA_Mutation-20160128</code></pre>
<pre><code>## see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation</code></pre>
<pre><code>## loading from cache</code></pre>
<pre><code>## Working on: BRCA_RNASeq2GeneNorm-20160128</code></pre>
<pre><code>## see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation</code></pre>
<pre><code>## loading from cache</code></pre>
<pre><code>## Working on: BRCA_Methylation_methyl27-20160128,
##  BRCA_Methylation_methyl450-20160128</code></pre>
<pre><code>## see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation</code></pre>
<pre><code>## loading from cache</code></pre>
<pre><code>## require("rhdf5")</code></pre>
<pre><code>## see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation</code></pre>
<pre><code>## loading from cache</code></pre>
<pre><code>## Working on: BRCA_Methylation_methyl27-20160128,
##  BRCA_Methylation_methyl450-20160128</code></pre>
<pre><code>## see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation</code></pre>
<pre><code>## loading from cache</code></pre>
<pre><code>## see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation</code></pre>
<pre><code>## loading from cache</code></pre>
<pre><code>## Loading required package: RaggedExperiment</code></pre>
<pre><code>## Working on: BRCA_colData-20160128</code></pre>
<pre><code>## see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation</code></pre>
<pre><code>## loading from cache</code></pre>
<pre><code>## Working on: BRCA_metadata-20160128</code></pre>
<pre><code>## see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation</code></pre>
<pre><code>## loading from cache</code></pre>
<pre><code>## Working on: BRCA_sampleMap-20160128</code></pre>
<pre><code>## see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation</code></pre>
<pre><code>## loading from cache</code></pre>
<pre><code>## harmonizing input:
##   removing 10940 sampleMap rows not in names(experiments)</code></pre>
<pre><code>##    user  system elapsed 
##  52.162   1.267  55.646</code></pre>
<p>We then add gene symbol annotations with genomic ranges and drop any unmapped symbols. We also do the same for CpG probe identifiers:</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1"></a>brca0 &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/TCGAutils/man/simplifyTCGA.html">symbolsToRanges</a></span>(brca, <span class="dt">unmapped =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<pre><code>## </code></pre>
<pre><code>## 'select()' returned 1:1 mapping between keys and columns</code></pre>
<pre><code>## harmonizing input:
##   removing 1212 sampleMap rows not in names(experiments)</code></pre>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a><span class="cf">if</span> (<span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span>(<span class="st">"IlluminaHumanMethylation450kanno.ilmn12.hg19"</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>))</span>
<span id="cb37-2"><a href="#cb37-2"></a>    BiocManager<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"IlluminaHumanMethylation450kanno.ilmn12.hg19"</span>)</span></code></pre></div>
<pre><code>## 
## Attaching package: 'Biostrings'</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     strsplit</code></pre>
<pre><code>## locfit 1.5-9.1    2013-03-22</code></pre>
<pre><code>## Setting options('download.file.method.GEOquery'='auto')</code></pre>
<pre><code>## Setting options('GEOquery.inmemory.gpl'=FALSE)</code></pre>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(IlluminaHumanMethylation450kanno.ilmn12.hg19)</span>
<span id="cb43-2"><a href="#cb43-2"></a></span>
<span id="cb43-3"><a href="#cb43-3"></a>brca1 &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/TCGAutils/man/simplifyTCGA.html">CpGtoRanges</a></span>(brca0, <span class="dt">unmapped =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<pre><code>## harmonizing input:
##   removing 1228 sampleMap rows not in names(experiments)</code></pre>
<p>Select only tumors in the data</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1"></a><span class="kw"><a href="https://rdrr.io/pkg/TCGAutils/man/curatedTCGAData-helpers.html">sampleTables</a></span>(brca1)</span></code></pre></div>
<pre><code>## $`BRCA_Mutation-20160128`
## 
##  01  06 
## 988   5 
## 
## $`BRCA_RNASeq2GeneNorm-20160128_ranged`
## 
##   01   06   11 
## 1093    7  112 
## 
## $`BRCA_Methylation_methyl27-20160128_ranged`
## 
##  01  06  11 
## 314   2  27 
## 
## $`BRCA_Methylation_methyl450-20160128_ranged`
## 
##  01  06  11 
## 783   5  97</code></pre>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1"></a><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(sampleTypes)</span></code></pre></div>
<pre><code>##   Code                                      Definition Short.Letter.Code
## 1   01                             Primary Solid Tumor                TP
## 2   02                           Recurrent Solid Tumor                TR
## 3   03 Primary Blood Derived Cancer - Peripheral Blood                TB
## 4   04    Recurrent Blood Derived Cancer - Bone Marrow              TRBM
## 5   05                        Additional - New Primary               TAP
## 6   06                                      Metastatic                TM</code></pre>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1"></a>sampleselector &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(brca1), <span class="cf">function</span>(x)</span>
<span id="cb49-2"><a href="#cb49-2"></a>    <span class="kw"><a href="https://rdrr.io/pkg/TCGAutils/man/TCGAsampleSelect.html">TCGAsampleSelect</a></span>(x, <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"01"</span>, <span class="st">"06"</span>))</span>
<span id="cb49-3"><a href="#cb49-3"></a>)</span>
<span id="cb49-4"><a href="#cb49-4"></a>brca1 &lt;-<span class="st"> </span>brca1[, sampleselector]</span></code></pre></div>
<pre><code>## harmonizing input:
##   removing 236 sampleMap rows with 'colname' not in colnames of experiments</code></pre>
<p>We check what probes “have at least a methylation…” across all samples:</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1"></a>meth.has.genes &lt;-</span>
<span id="cb51-2"><a href="#cb51-2"></a><span class="st">    </span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="kw">rowData</span>(</span>
<span id="cb51-3"><a href="#cb51-3"></a>        brca1[[<span class="st">"BRCA_Methylation_methyl450-20160128_ranged"</span>]])<span class="op">$</span>Gene_Symbol)</span>
<span id="cb51-4"><a href="#cb51-4"></a>brca1 &lt;-</span>
<span id="cb51-5"><a href="#cb51-5"></a><span class="st">    </span><span class="kw"><a href="https://rdrr.io/pkg/MultiAssayExperiment/man/subsetBy.html">subsetByRow</a></span>(brca1,</span>
<span id="cb51-6"><a href="#cb51-6"></a>        <span class="kw">List</span>(<span class="st">"BRCA_Methylation_methyl450-20160128_ranged"</span>=meth.has.genes))</span>
<span id="cb51-7"><a href="#cb51-7"></a>genes &lt;-</span>
<span id="cb51-8"><a href="#cb51-8"></a><span class="st">    </span><span class="kw">rowData</span>(brca1[[<span class="st">"BRCA_Methylation_methyl450-20160128_ranged"</span>]])<span class="op">$</span>Gene_Symbol</span>
<span id="cb51-9"><a href="#cb51-9"></a>methylated &lt;-<span class="st"> </span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span>(</span>
<span id="cb51-10"><a href="#cb51-10"></a>    <span class="kw">assay</span>(brca1[[<span class="st">"BRCA_Methylation_methyl450-20160128_ranged"</span>]])</span>
<span id="cb51-11"><a href="#cb51-11"></a>))</span></code></pre></div>
<p>Number of non-missing per gene in each column</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1"></a><span class="co"># convert to integer first</span></span>
<span id="cb52-2"><a href="#cb52-2"></a>meth &lt;-<span class="st"> </span>methylated <span class="op">*</span><span class="st"> </span>1L</span>
<span id="cb52-3"><a href="#cb52-3"></a><span class="kw"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(meth1 &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/rowsum.html">rowsum</a></span>(meth, <span class="dt">group =</span> genes))</span></code></pre></div>
<pre><code>##    user  system elapsed 
##   5.271   0.040   5.311</code></pre>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1"></a><span class="co"># where genes have at least one methylation</span></span>
<span id="cb54-2"><a href="#cb54-2"></a>meth2 &lt;-<span class="st"> </span>meth1 <span class="op">&gt;</span><span class="st"> </span>0L</span>
<span id="cb54-3"><a href="#cb54-3"></a>brca1 &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(brca1, <span class="dt">has.meth =</span> <span class="kw">SummarizedExperiment</span>(meth2))</span></code></pre></div>
<p>Get ranges and add “within the first 2000 bp outside of the genes”</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1"></a>rnaranges &lt;-<span class="st"> </span><span class="kw">rowRanges</span>(brca1[[<span class="st">"BRCA_RNASeq2GeneNorm-20160128_ranged"</span>]])</span>
<span id="cb55-2"><a href="#cb55-2"></a>allROIs &lt;-<span class="st"> </span><span class="kw">flank</span>(rnaranges, <span class="dv">2000</span>, <span class="dt">both =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p>Find any somatic mutations within regions of interest:</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1"></a>somatic &lt;-<span class="st"> </span><span class="cf">function</span>(scores, ranges, qranges) <span class="kw"><a href="https://rdrr.io/r/base/any.html">any</a></span>(<span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(scores))</span>
<span id="cb56-2"><a href="#cb56-2"></a></span>
<span id="cb56-3"><a href="#cb56-3"></a><span class="kw">genome</span>(brca1[[<span class="st">"BRCA_Mutation-20160128"</span>]]) &lt;-</span>
<span id="cb56-4"><a href="#cb56-4"></a><span class="st">    </span>TCGAutils<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/TCGAutils/man/builds.html">translateBuild</a></span>(<span class="kw">genome</span>(brca1[[<span class="st">"BRCA_Mutation-20160128"</span>]]))</span>
<span id="cb56-5"><a href="#cb56-5"></a></span>
<span id="cb56-6"><a href="#cb56-6"></a>mutations &lt;-<span class="st"> </span>RaggedExperiment<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/RaggedExperiment/man/assay-functions.html">qreduceAssay</a></span>(</span>
<span id="cb56-7"><a href="#cb56-7"></a>    <span class="dt">x =</span> brca1[[<span class="st">"BRCA_Mutation-20160128"</span>]],</span>
<span id="cb56-8"><a href="#cb56-8"></a>    <span class="dt">query =</span> allROIs,</span>
<span id="cb56-9"><a href="#cb56-9"></a>    <span class="dt">simplifyReduce =</span> somatic,</span>
<span id="cb56-10"><a href="#cb56-10"></a>    <span class="dt">i =</span> <span class="st">"Variant_Classification"</span></span>
<span id="cb56-11"><a href="#cb56-11"></a>)</span>
<span id="cb56-12"><a href="#cb56-12"></a><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(mutations) &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(rnaranges)</span>
<span id="cb56-13"><a href="#cb56-13"></a>mutations[<span class="kw"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(mutations)] &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb56-14"><a href="#cb56-14"></a></span>
<span id="cb56-15"><a href="#cb56-15"></a>brca1 &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(brca1, <span class="dt">mutations =</span> <span class="kw">SummarizedExperiment</span>(mutations))</span></code></pre></div>
<p>Handle technical replicates and take the sum of number of mutations per sample</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1"></a>reps &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/MultiAssayExperiment/man/MultiAssayExperiment-helpers.html">replicated</a></span>(brca1)[[<span class="st">"mutations"</span>]]</span>
<span id="cb57-2"><a href="#cb57-2"></a>compreps &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/MultiAssayExperiment/man/MultiAssayExperiment-helpers.html">mergeReplicates</a></span>(<span class="kw">assay</span>(brca1[[<span class="st">"mutations"</span>]]), reps, sum)</span>
<span id="cb57-3"><a href="#cb57-3"></a>brca1[[<span class="st">"mutations"</span>]] &lt;-<span class="st"> </span><span class="kw">SummarizedExperiment</span>(compreps)</span></code></pre></div>
<pre><code>## harmonizing input:
##   removing 16 sampleMap rows with 'colname' not in colnames of experiments</code></pre>
<p>Resolving replicates in other assays</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1"></a>brca2 &lt;-<span class="st"> </span>brca1[, ,</span>
<span id="cb59-2"><a href="#cb59-2"></a>    <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"BRCA_RNASeq2GeneNorm-20160128_ranged"</span>, <span class="st">"has.meth"</span>, <span class="st">"mutations"</span>)]</span></code></pre></div>
<pre><code>## harmonizing input:
##   removing 2097 sampleMap rows not in names(experiments)</code></pre>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1"></a>rnamerged &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/MultiAssayExperiment/man/MultiAssayExperiment-helpers.html">mergeReplicates</a></span>(brca2[[<span class="st">"BRCA_RNASeq2GeneNorm-20160128_ranged"</span>]],</span>
<span id="cb61-2"><a href="#cb61-2"></a>    <span class="dt">replicates =</span> <span class="kw"><a href="https://rdrr.io/pkg/MultiAssayExperiment/man/MultiAssayExperiment-helpers.html">replicated</a></span>(brca2[, , <span class="st">"BRCA_RNASeq2GeneNorm-20160128_ranged"</span>])[[1L]],</span>
<span id="cb61-3"><a href="#cb61-3"></a>    <span class="dt">simplify =</span> <span class="cf">function</span> (x) {</span>
<span id="cb61-4"><a href="#cb61-4"></a>        gt3 &lt;-<span class="st"> </span>x <span class="op">&gt;</span><span class="st"> </span><span class="dv">3</span></span>
<span id="cb61-5"><a href="#cb61-5"></a>        <span class="cf">if</span> (<span class="kw"><a href="https://rdrr.io/r/base/any.html">any</a></span>(gt3))</span>
<span id="cb61-6"><a href="#cb61-6"></a>            BiocGenerics<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html">mean</a></span>(x[gt3])</span>
<span id="cb61-7"><a href="#cb61-7"></a>        <span class="cf">else</span></span>
<span id="cb61-8"><a href="#cb61-8"></a>            x[[<span class="dv">1</span>]]</span>
<span id="cb61-9"><a href="#cb61-9"></a>    }</span>
<span id="cb61-10"><a href="#cb61-10"></a>)</span></code></pre></div>
<pre><code>## harmonizing input:
##   removing 1765 sampleMap rows not in names(experiments)
##   removing 5 colData rownames not in sampleMap 'primary'</code></pre>
<pre><code>## Warning in if (any(gt3)) BiocGenerics::mean(x[gt3]) else x[[1]]: the condition
## has length &gt; 1 and only the first element will be used

## Warning in if (any(gt3)) BiocGenerics::mean(x[gt3]) else x[[1]]: the condition
## has length &gt; 1 and only the first element will be used

## Warning in if (any(gt3)) BiocGenerics::mean(x[gt3]) else x[[1]]: the condition
## has length &gt; 1 and only the first element will be used

## Warning in if (any(gt3)) BiocGenerics::mean(x[gt3]) else x[[1]]: the condition
## has length &gt; 1 and only the first element will be used

## Warning in if (any(gt3)) BiocGenerics::mean(x[gt3]) else x[[1]]: the condition
## has length &gt; 1 and only the first element will be used

## Warning in if (any(gt3)) BiocGenerics::mean(x[gt3]) else x[[1]]: the condition
## has length &gt; 1 and only the first element will be used

## Warning in if (any(gt3)) BiocGenerics::mean(x[gt3]) else x[[1]]: the condition
## has length &gt; 1 and only the first element will be used</code></pre>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1"></a>brca2[[<span class="st">"BRCA_RNASeq2GeneNorm-20160128_ranged"</span>]] &lt;-<span class="st"> </span>rnamerged</span></code></pre></div>
<pre><code>## harmonizing input:
##   removing 7 sampleMap rows with 'colname' not in colnames of experiments</code></pre>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1"></a>methmerge &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/MultiAssayExperiment/man/MultiAssayExperiment-helpers.html">mergeReplicates</a></span>(brca2[[<span class="st">"has.meth"</span>]],</span>
<span id="cb66-2"><a href="#cb66-2"></a>    <span class="dt">replicates =</span> <span class="kw"><a href="https://rdrr.io/pkg/MultiAssayExperiment/man/MultiAssayExperiment-helpers.html">replicated</a></span>(brca2[, , <span class="st">"has.meth"</span>])[[1L]],</span>
<span id="cb66-3"><a href="#cb66-3"></a>    <span class="dt">simplify =</span> any</span>
<span id="cb66-4"><a href="#cb66-4"></a>)</span></code></pre></div>
<pre><code>## harmonizing input:
##   removing 2070 sampleMap rows not in names(experiments)
##   removing 314 colData rownames not in sampleMap 'primary'</code></pre>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1"></a>mutmerge &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/MultiAssayExperiment/man/MultiAssayExperiment-helpers.html">mergeReplicates</a></span>(brca2[[<span class="st">"mutations"</span>]],</span>
<span id="cb68-2"><a href="#cb68-2"></a>    <span class="dt">replicates =</span> <span class="kw"><a href="https://rdrr.io/pkg/MultiAssayExperiment/man/MultiAssayExperiment-helpers.html">replicated</a></span>(brca2[, , <span class="st">"mutations"</span>])[[1L]],</span>
<span id="cb68-3"><a href="#cb68-3"></a>    <span class="dt">simplify =</span> sum</span>
<span id="cb68-4"><a href="#cb68-4"></a>)</span></code></pre></div>
<pre><code>## harmonizing input:
##   removing 1881 sampleMap rows not in names(experiments)
##   removing 121 colData rownames not in sampleMap 'primary'</code></pre>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1"></a>brca2[[<span class="st">"has.meth"</span>]] &lt;-<span class="st"> </span>methmerge</span></code></pre></div>
<pre><code>## harmonizing input:
##   removing 4 sampleMap rows with 'colname' not in colnames of experiments</code></pre>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1"></a>brca2[[<span class="st">"mutations"</span>]] &lt;-<span class="st"> </span>mutmerge</span>
<span id="cb72-2"><a href="#cb72-2"></a></span>
<span id="cb72-3"><a href="#cb72-3"></a>brca3 &lt;-<span class="st"> </span><span class="kw">as</span>(brca2, <span class="st">"MatchedAssayExperiment"</span>)</span>
<span id="cb72-4"><a href="#cb72-4"></a>brca3 &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/MultiAssayExperiment/man/MultiAssayExperiment-helpers.html">intersectRows</a></span>(brca3)</span></code></pre></div>
<p>An expression threshold is used where “FPKM &gt; 3”. In our case, we use the better alternative: TPM &gt; 3</p>
<p><a href="https://haroldpimentel.wordpress.com/2014/05/08/what-the-fpkm-a-review-rna-seq-expression-units/" class="uri">https://haroldpimentel.wordpress.com/2014/05/08/what-the-fpkm-a-review-rna-seq-expression-units/</a> <a href="http://diytranscriptomics.com/Reading/files/wagnerTPM.pdf" class="uri">http://diytranscriptomics.com/Reading/files/wagnerTPM.pdf</a></p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1"></a>keep &lt;-<span class="st"> </span><span class="kw">assay</span>(brca3, <span class="st">"BRCA_RNASeq2GeneNorm-20160128_ranged"</span>) <span class="op">&gt;</span><span class="st"> </span><span class="dv">3</span> <span class="op">&amp;</span></span>
<span id="cb73-2"><a href="#cb73-2"></a><span class="st">  </span><span class="kw">assay</span>(brca3, <span class="st">"has.meth"</span>) <span class="op">&amp;</span></span>
<span id="cb73-3"><a href="#cb73-3"></a><span class="st">  </span><span class="kw">assay</span>(brca3, <span class="st">"mutations"</span>)</span>
<span id="cb73-4"><a href="#cb73-4"></a></span>
<span id="cb73-5"><a href="#cb73-5"></a><span class="kw"><a href="https://rdrr.io/r/base/summary.html">summary</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span>(keep))</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
##   0.000   0.000   1.000   2.934   3.000 157.000       2</code></pre>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1"></a><span class="kw"><a href="https://rdrr.io/r/base/sort.html">sort</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span>(keep), <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)[<span class="kw"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(keep)<span class="op">*</span><span class="fl">0.05</span>))]</span></code></pre></div>
<pre><code>## TCGA-AC-A23H-01A-11R-A157-07 TCGA-A2-A0T5-01A-21R-A084-07 
##                          157                           59 
## TCGA-A8-A0A6-01A-12R-A056-07 TCGA-BH-A0B6-01A-11R-A19W-07 
##                           59                           40 
## TCGA-AO-A128-01A-11R-A10J-07 TCGA-AC-A5XS-01A-11R-A29R-07 
##                           37                           29 
## TCGA-D8-A27G-01A-11R-A16F-07 TCGA-AO-A03M-01B-11R-A10J-07 
##                           29                           27 
## TCGA-D8-A1JA-01A-11R-A13Q-07 TCGA-BH-A0HF-01A-11R-A056-07 
##                           27                           26 
## TCGA-D8-A1XK-01A-21R-A14M-07 TCGA-C8-A26Y-01A-11R-A16F-07 
##                           26                           25 
## TCGA-D8-A1XQ-01A-11R-A14M-07 TCGA-D8-A1J8-01A-11R-A13Q-07 
##                           21                           18 
## TCGA-PE-A5DE-01A-11R-A27Q-07 TCGA-AO-A12E-01A-11R-A10J-07 
##                           18                           16 
## TCGA-EW-A1IZ-01A-11R-A13Q-07 TCGA-BH-A2L8-01A-11R-A18M-07 
##                           15                           14 
## TCGA-AR-A2LE-01A-11R-A180-07 TCGA-AC-A3W6-01A-12R-A22K-07 
##                           13                           12 
## TCGA-C8-A1HM-01A-12R-A137-07 TCGA-C8-A274-01A-11R-A16F-07 
##                           12                           12 
## TCGA-A7-A56D-01A-11R-A27Q-07 TCGA-A7-A5ZV-01A-11R-A28M-07 
##                           12                           12 
## TCGA-A1-A0SI-01A-11R-A144-07 TCGA-GM-A2D9-01A-11R-A18M-07 
##                           11                           11 
## TCGA-AC-A5EH-01A-11R-A28M-07 TCGA-A7-A26E-01A-11R-A169-07 
##                           11                           10 
## TCGA-A7-A26J-01A-11R-A169-07 TCGA-AC-A3TN-01A-11R-A22K-07 
##                           10                           10 
## TCGA-BH-A0HA-01A-11R-A12P-07 TCGA-E2-A15I-01A-21R-A137-07 
##                           10                           10 
## TCGA-A2-A3Y0-01A-11R-A239-07 
##                           10</code></pre>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Marcel Ramos, Levi Waldron.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
