<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Presents example analyses using curatedTCGAData, MultiAssayExperiment,
  and TCGAutils • curatedTCGAManu</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Presents example analyses using curatedTCGAData, MultiAssayExperiment,
  and TCGAutils">
<meta property="og:description" content='This package reproduces all results presented in the manuscript 
  "Multi-omic integration of public oncology databases in Bioconductor." '>
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">curatedTCGAManu</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="articles/Figure2_OncoPrint.html">PanCan OncoPrint</a>
</li>
<li>
  <a href="articles/Figures3n4_GSEA_PanCan.html">PanCan DE &amp; GSEA</a>
</li>
<li>
  <a href="articles/Figures5-6-S4.html">Multi-omic Plots</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/LiNk-NY/curatedTCGAManu/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div id="curatedtcgamanu" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#curatedtcgamanu" class="anchor"></a>curatedTCGAManu</h1></div>
<div id="multi-omic-integration-of-public-oncology-databases-in-bioconductor" class="section level2">
<h2 class="hasAnchor">
<a href="#multi-omic-integration-of-public-oncology-databases-in-bioconductor" class="anchor"></a>Multi-omic integration of public oncology databases in Bioconductor</h2>

<p>This repository contains scripts and datasets for the curatedTCGAData + cBioPortalData manuscript.</p>
</div>
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p>We provide several examples to demonstrate the powerful and flexible analysis environment provided. These analyses, previously only achievable through a significant investment of time and bioinformatic training, become straight- forward analysis exercises provided as vignettes.</p>
</div>
<div id="key-points" class="section level2">
<h2 class="hasAnchor">
<a href="#key-points" class="anchor"></a>Key Points</h2>
<ul>
<li><p>Key objective: To provide flexible, integrated multi-omic representations of public oncology databases in R/Bioconductor with grealy reduced data management overhead.</p></li>
<li><p>Knowledge generated: Our Bioconductor software packages provide a novel approach to lower barriers to analysis and tool development for the TCGA and cBioPortal databases.</p></li>
<li><p>Relevance: Our tools provide flexible, programmatic analysis of hundreds of fully integrated multi’omic oncology datasets within an ecosystem of multi-omic analysis tools.</p></li>
</ul>
</div>
<div id="key-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#key-packages" class="anchor"></a>Key Packages</h2>
<ul>
<li><a href="http://waldronlab.io/MultiAssayExperiment/index.html">MultiAssayExperiment</a></li>
<li><a href="https://bioconductor.org/packages/curatedTCGAData">curatedTCGAData</a></li>
<li><a href="https://bioconductor.org/packages/TCGAutils">TCGAutils</a></li>
<li><a href="https://github.com/waldronlab/cBioPortalData">cBioPortalData</a></li>
<li><a href="https://bioconductor.org/packages/EnrichmentBrowser">EnrichmentBrowser</a></li>
<li><a href="https://bioconductor.org/packages/GSEABenchmarkeR">GSEABenchmarkeR</a></li>
<li><a href="https://bioconductor.org/packages/RTCGAToolbox">RTCGAToolbox</a></li>
</ul>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>Until the release of Bioconductor 3.11 (scheduled for April 28, 2020), it is strongly recommended to use the <em>devel</em> version of Bioconductor. That version can be installed the <a href="https://bioconductor.org/developers/how-to/useDevel/">traditional way</a> or by using the <a href="https://bioconductor.org/help/docker/">Docker container</a>.</p>
<p>Additionally until the release of Bioconductor 3.11, <code>cBioPortalData</code> must be installed from GitHub as shown in the following code chunk which installs all necessary packages either directly or as dependencies. Note that this code chunk is not evaluated, because installation only needs to be performed once.</p>
<div class="sourceCode" id="cb1"><pre class="r"><span class="kw pkg">BiocManager</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"cBioPortalData"</span>, <span class="st">"LiNk-NY/curatedTCGAManu"</span>))</pre></div>
</div>
<div id="vignette-build" class="section level2">
<h2 class="hasAnchor">
<a href="#vignette-build" class="anchor"></a>Vignette Build</h2>
<p>Because of the size of the data, it is recommended that the vignettes be built individually. If using RStudio, the user can simply open the vignette and press the <code>knit</code> button. Otherwise, the package can be built completely with vignettes by doing:</p>
<pre><code>    R CMD build curatedTCGAManu</code></pre>
<p>in the command line or</p>
<div class="sourceCode" id="cb3"><pre class="r"><span class="kw pkg">BiocManager</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"Link-NY/curatedTCGAManu"</span>, <span class="kw">build_vignettes</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></div>
<p>in R.</p>
</div>
<div id="loading-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#loading-packages" class="anchor"></a>Loading packages</h2>
<div class="sourceCode" id="cb4"><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">curatedTCGAData</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">cBioPortalData</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">TCGAutils</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">GenomicDataCommons</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">rtracklayer</span>)</pre></div>
<p><em>Note</em>. For clarity, we include <code>library</code> commands within the supplemental code chunks.</p>
</div>
</div>
<div id="figures" class="section level1">
<h1 class="hasAnchor">
<a href="#figures" class="anchor"></a>Figures</h1>
<div id="ease-of-use-schematic" class="section level2">
<h2 class="hasAnchor">
<a href="#ease-of-use-schematic" class="anchor"></a>Ease-of-use schematic</h2>
<div id="figure-1" class="section level3">
<h3 class="hasAnchor">
<a href="#figure-1" class="anchor"></a>Figure 1</h3>
<p><a href="https://github.com/waldronlab/schematics/"><img src="https://raw.githubusercontent.com/waldronlab/schematics/master/pngs/F1_curatedTCGAData.png"></a></p>
</div>
</div>
<div id="pan-cancer-oncoprint-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#pan-cancer-oncoprint-plot" class="anchor"></a>Pan-Cancer OncoPrint Plot</h2>
<div id="figure-2" class="section level3">
<h3 class="hasAnchor">
<a href="#figure-2" class="anchor"></a>Figure 2</h3>
<p><a href="https://github.com/waldronlab/schematics/"><img src="https://raw.githubusercontent.com/waldronlab/schematics/master/pngs/unnamed-chunk-20-1.png"></a></p>
</div>
</div>
<div id="differential-expression-and-gsea-pancan" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-expression-and-gsea-pancan" class="anchor"></a>Differential Expression and GSEA PanCan</h2>
<div id="figure-3" class="section level3">
<h3 class="hasAnchor">
<a href="#figure-3" class="anchor"></a>Figure 3</h3>
<p><a href="https://github.com/waldronlab/schematics/"><img src="https://raw.githubusercontent.com/waldronlab/schematics/master/pngs/unnamed-chunk-17-1.png"></a></p>
</div>
<div id="figure-4" class="section level3">
<h3 class="hasAnchor">
<a href="#figure-4" class="anchor"></a>Figure 4</h3>
<p><a href="https://github.com/waldronlab/schematics/"><img src="https://raw.githubusercontent.com/waldronlab/schematics/master/pngs/unnamed-chunk-24-1.png"></a></p>
</div>
</div>
<div id="example-multi-omic-analyses" class="section level2">
<h2 class="hasAnchor">
<a href="#example-multi-omic-analyses" class="anchor"></a>Example Multi-omic Analyses</h2>
<div id="figure-5" class="section level3">
<h3 class="hasAnchor">
<a href="#figure-5" class="anchor"></a>Figure 5</h3>
<p><a href="https://github.com/waldronlab/schematics/"><img src="https://raw.githubusercontent.com/waldronlab/schematics/master/pngs/unnamed-chunk-9-1.png"></a></p>
</div>
<div id="figure-6" class="section level3">
<h3 class="hasAnchor">
<a href="#figure-6" class="anchor"></a>Figure 6</h3>
<p><a href="https://github.com/waldronlab/schematics/"><img src="https://raw.githubusercontent.com/waldronlab/schematics/master/pngs/unnamed-chunk-14-1.png"></a></p>
</div>
</div>
</div>
<div id="supplement-reference" class="section level1">
<h1 class="hasAnchor">
<a href="#supplement-reference" class="anchor"></a>Supplement Reference</h1>
<div id="figure-s1" class="section level2">
<h2 class="hasAnchor">
<a href="#figure-s1" class="anchor"></a>Figure S1</h2>
<div id="data-provenance-and-package-network" class="section level3">
<h3 class="hasAnchor">
<a href="#data-provenance-and-package-network" class="anchor"></a>Data provenance and package network</h3>
<p><a href="https://github.com/waldronlab/schematics/"><img src="https://raw.githubusercontent.com/waldronlab/schematics/master/pngs/TCGAMAEPipelineSchematic.png"></a></p>
</div>
</div>
<div id="figure-s2a" class="section level2">
<h2 class="hasAnchor">
<a href="#figure-s2a" class="anchor"></a>Figure S2A</h2>
<div id="example-code-for-installing-and-downloading-tcga-data-using-curatedtcgadata" class="section level3">
<h3 class="hasAnchor">
<a href="#example-code-for-installing-and-downloading-tcga-data-using-curatedtcgadata" class="anchor"></a>Example code for installing and downloading TCGA data using curatedTCGAData</h3>
<div class="sourceCode" id="cb5"><pre class="r"><span class="kw">if</span> (!<span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, <span class="kw">quietly</span> <span class="kw">=</span> <span class="fl">TRUE</span>))
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)

<span class="kw">if</span> (!<span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"curatedTCGAData"</span>, <span class="kw">quietly</span> <span class="kw">=</span> <span class="fl">TRUE</span>))
    <span class="kw pkg">BiocManager</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"curatedTCGAData"</span>)

<span class="co">## Glioblastoma Multiforme (GBM)</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">curatedTCGAData</span>)
<span class="fu"><a href="https://rdrr.io/pkg/curatedTCGAData/man/curatedTCGAData.html">curatedTCGAData</a></span>(<span class="kw">diseaseCode</span> <span class="kw">=</span> <span class="st">"GBM"</span>, <span class="kw">assays</span> <span class="kw">=</span> <span class="st">"RNA*"</span>, <span class="kw">dry.run</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="co">#&gt; A MultiAssayExperiment object of 1 listed</span>
<span class="co">#&gt;  experiment with a user-defined name and respective class.</span>
<span class="co">#&gt;  Containing an ExperimentList class object of length 1:</span>
<span class="co">#&gt;  [1] GBM_RNASeq2GeneNorm-20160128: SummarizedExperiment with 20501 rows and 166 columns</span>
<span class="co">#&gt; Functionality:</span>
<span class="co">#&gt;  experiments() - obtain the ExperimentList instance</span>
<span class="co">#&gt;  colData() - the primary/phenotype DataFrame</span>
<span class="co">#&gt;  sampleMap() - the sample coordination DataFrame</span>
<span class="co">#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span>
<span class="co">#&gt;  *Format() - convert into a long or wide DataFrame</span>
<span class="co">#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</span>
<span class="co">#&gt;  exportClass() - save all data to files</span></pre></div>
</div>
</div>
<div id="figure-s2b" class="section level2">
<h2 class="hasAnchor">
<a href="#figure-s2b" class="anchor"></a>Figure S2B</h2>
<div id="example-cbioportaldata-code-for-downloading-and-exporting-tcga-data-from-cbioportalorg-and-via-the-cbioportal-api" class="section level3">
<h3 class="hasAnchor">
<a href="#example-cbioportaldata-code-for-downloading-and-exporting-tcga-data-from-cbioportalorg-and-via-the-cbioportal-api" class="anchor"></a>Example cBioPortalData code for downloading and exporting TCGA data from cBioPortal.org and via the cBioPortal API</h3>
<div class="sourceCode" id="cb6"><pre class="r"><span class="co">## installation</span>
<span class="kw">if</span> (!<span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"cBioPortalData"</span>, <span class="kw">quietly</span> <span class="kw">=</span> <span class="fl">TRUE</span>))
    <span class="kw pkg">BiocManager</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"cBioPortalData"</span>)

<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">cBioPortalData</span>)

<span class="co">## https://cbioportal.org/datasets (Bulk data method)</span>
<span class="no">gbm</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/cBioPortalData/man/cBioDataPack.html">cBioDataPack</a></span>(<span class="st">"gbm_tcga"</span>)

<span class="co">## https://cBioPortal.org/api (API method)</span>
<span class="no">cBio</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/cBioPortalData/man/cBioPortal.html">cBioPortal</a></span>()

<span class="co">## use exportClass() with the result to save data files</span>
<span class="no">tcga_gbm</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/cBioPortalData/man/cBioPortalData.html">cBioPortalData</a></span>(<span class="no">cBio</span>, <span class="kw">studyId</span> <span class="kw">=</span> <span class="st">"gbm_tcga"</span>, <span class="kw">genePanelId</span> <span class="kw">=</span> <span class="st">"IMPACT341"</span>)

<span class="no">tcga_gbm</span>
<span class="co">#&gt; A MultiAssayExperiment object of 16 listed</span>
<span class="co">#&gt;  experiments with user-defined names and respective classes.</span>
<span class="co">#&gt;  Containing an ExperimentList class object of length 16:</span>
<span class="co">#&gt;  [1] gbm_tcga_rppa: SummarizedExperiment with 67 rows and 244 columns</span>
<span class="co">#&gt;  [2] gbm_tcga_rppa_Zscores: SummarizedExperiment with 67 rows and 244 columns</span>
<span class="co">#&gt;  [3] gbm_tcga_gistic: SummarizedExperiment with 339 rows and 577 columns</span>
<span class="co">#&gt;  [4] gbm_tcga_mrna_U133: SummarizedExperiment with 311 rows and 528 columns</span>
<span class="co">#&gt;  [5] gbm_tcga_mrna_U133_Zscores: SummarizedExperiment with 308 rows and 528 columns</span>
<span class="co">#&gt;  [6] gbm_tcga_mrna: SummarizedExperiment with 334 rows and 401 columns</span>
<span class="co">#&gt;  [7] gbm_tcga_mrna_median_Zscores: SummarizedExperiment with 329 rows and 401 columns</span>
<span class="co">#&gt;  [8] gbm_tcga_rna_seq_v2_mrna: SummarizedExperiment with 341 rows and 166 columns</span>
<span class="co">#&gt;  [9] gbm_tcga_rna_seq_v2_mrna_median_Zscores: SummarizedExperiment with 333 rows and 166 columns</span>
<span class="co">#&gt;  [10] gbm_tcga_linear_CNA: SummarizedExperiment with 339 rows and 577 columns</span>
<span class="co">#&gt;  [11] gbm_tcga_methylation_hm27: SummarizedExperiment with 282 rows and 285 columns</span>
<span class="co">#&gt;  [12] gbm_tcga_methylation_hm450: SummarizedExperiment with 282 rows and 153 columns</span>
<span class="co">#&gt;  [13] gbm_tcga_mutations: RangedSummarizedExperiment with 810 rows and 271 columns</span>
<span class="co">#&gt;  [14] gbm_tcga_rna_seq_v2_mrna_median_all_sample_Zscores: SummarizedExperiment with 340 rows and 166 columns</span>
<span class="co">#&gt;  [15] gbm_tcga_mrna_median_all_sample_Zscores: SummarizedExperiment with 304 rows and 401 columns</span>
<span class="co">#&gt;  [16] gbm_tcga_mrna_U133_all_sample_Zscores: SummarizedExperiment with 311 rows and 528 columns</span>
<span class="co">#&gt; Functionality:</span>
<span class="co">#&gt;  experiments() - obtain the ExperimentList instance</span>
<span class="co">#&gt;  colData() - the primary/phenotype DataFrame</span>
<span class="co">#&gt;  sampleMap() - the sample coordination DataFrame</span>
<span class="co">#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span>
<span class="co">#&gt;  *Format() - convert into a long or wide DataFrame</span>
<span class="co">#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</span>
<span class="co">#&gt;  exportClass() - save all data to files</span>

<span class="fu">exportClass</span>(<span class="no">tcga_gbm</span>, <span class="kw">dir</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span>(), <span class="kw">fmt</span> <span class="kw">=</span> <span class="st">"csv"</span>)
<span class="co">#&gt;  [1] "/tmp/Rtmp1EL92I/tcga_gbm_META_1.csv"                                            </span>
<span class="co">#&gt;  [2] "/tmp/Rtmp1EL92I/tcga_gbm_META_2.csv"                                            </span>
<span class="co">#&gt;  [3] "/tmp/Rtmp1EL92I/tcga_gbm_META_3.csv"                                            </span>
<span class="co">#&gt;  [4] "/tmp/Rtmp1EL92I/tcga_gbm_META_4.csv"                                            </span>
<span class="co">#&gt;  [5] "/tmp/Rtmp1EL92I/tcga_gbm_META_5.csv"                                            </span>
<span class="co">#&gt;  [6] "/tmp/Rtmp1EL92I/tcga_gbm_META_6.csv"                                            </span>
<span class="co">#&gt;  [7] "/tmp/Rtmp1EL92I/tcga_gbm_META_7.csv"                                            </span>
<span class="co">#&gt;  [8] "/tmp/Rtmp1EL92I/tcga_gbm_META_8.csv"                                            </span>
<span class="co">#&gt;  [9] "/tmp/Rtmp1EL92I/tcga_gbm_META_9.csv"                                            </span>
<span class="co">#&gt; [10] "/tmp/Rtmp1EL92I/tcga_gbm_META_10.csv"                                           </span>
<span class="co">#&gt; [11] "/tmp/Rtmp1EL92I/tcga_gbm_META_11.csv"                                           </span>
<span class="co">#&gt; [12] "/tmp/Rtmp1EL92I/tcga_gbm_META_12.csv"                                           </span>
<span class="co">#&gt; [13] "/tmp/Rtmp1EL92I/tcga_gbm_META_13.csv"                                           </span>
<span class="co">#&gt; [14] "/tmp/Rtmp1EL92I/tcga_gbm_META_14.csv"                                           </span>
<span class="co">#&gt; [15] "/tmp/Rtmp1EL92I/tcga_gbm_META_15.csv"                                           </span>
<span class="co">#&gt; [16] "/tmp/Rtmp1EL92I/tcga_gbm_META_16.csv"                                           </span>
<span class="co">#&gt; [17] "/tmp/Rtmp1EL92I/tcga_gbm_gbm_tcga_rppa.csv"                                     </span>
<span class="co">#&gt; [18] "/tmp/Rtmp1EL92I/tcga_gbm_gbm_tcga_rppa_Zscores.csv"                             </span>
<span class="co">#&gt; [19] "/tmp/Rtmp1EL92I/tcga_gbm_gbm_tcga_gistic.csv"                                   </span>
<span class="co">#&gt; [20] "/tmp/Rtmp1EL92I/tcga_gbm_gbm_tcga_mrna_U133.csv"                                </span>
<span class="co">#&gt; [21] "/tmp/Rtmp1EL92I/tcga_gbm_gbm_tcga_mrna_U133_Zscores.csv"                        </span>
<span class="co">#&gt; [22] "/tmp/Rtmp1EL92I/tcga_gbm_gbm_tcga_mrna.csv"                                     </span>
<span class="co">#&gt; [23] "/tmp/Rtmp1EL92I/tcga_gbm_gbm_tcga_mrna_median_Zscores.csv"                      </span>
<span class="co">#&gt; [24] "/tmp/Rtmp1EL92I/tcga_gbm_gbm_tcga_rna_seq_v2_mrna.csv"                          </span>
<span class="co">#&gt; [25] "/tmp/Rtmp1EL92I/tcga_gbm_gbm_tcga_rna_seq_v2_mrna_median_Zscores.csv"           </span>
<span class="co">#&gt; [26] "/tmp/Rtmp1EL92I/tcga_gbm_gbm_tcga_linear_CNA.csv"                               </span>
<span class="co">#&gt; [27] "/tmp/Rtmp1EL92I/tcga_gbm_gbm_tcga_methylation_hm27.csv"                         </span>
<span class="co">#&gt; [28] "/tmp/Rtmp1EL92I/tcga_gbm_gbm_tcga_methylation_hm450.csv"                        </span>
<span class="co">#&gt; [29] "/tmp/Rtmp1EL92I/tcga_gbm_gbm_tcga_mutations.csv"                                </span>
<span class="co">#&gt; [30] "/tmp/Rtmp1EL92I/tcga_gbm_gbm_tcga_rna_seq_v2_mrna_median_all_sample_Zscores.csv"</span>
<span class="co">#&gt; [31] "/tmp/Rtmp1EL92I/tcga_gbm_gbm_tcga_mrna_median_all_sample_Zscores.csv"           </span>
<span class="co">#&gt; [32] "/tmp/Rtmp1EL92I/tcga_gbm_gbm_tcga_mrna_U133_all_sample_Zscores.csv"             </span>
<span class="co">#&gt; [33] "/tmp/Rtmp1EL92I/tcga_gbm_colData.csv"                                           </span>
<span class="co">#&gt; [34] "/tmp/Rtmp1EL92I/tcga_gbm_sampleMap.csv"</span></pre></div>
</div>
</div>
<div id="figure-s2c" class="section level2">
<h2 class="hasAnchor">
<a href="#figure-s2c" class="anchor"></a>Figure S2C</h2>
<div id="example-hg19-to-hg38-liftover-procedure-using-bioconductor-tools" class="section level3">
<h3 class="hasAnchor">
<a href="#example-hg19-to-hg38-liftover-procedure-using-bioconductor-tools" class="anchor"></a>Example hg19 to hg38 liftover procedure using Bioconductor tools</h3>
<div class="sourceCode" id="cb7"><pre class="r"><span class="no">liftchain</span> <span class="kw">&lt;-</span> <span class="st">"http://hgdownload.cse.ucsc.edu/goldenpath/hg19/liftOver/hg19ToHg38.over.chain.gz"</span>
<span class="no">cloc38</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span>(), <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>(<span class="st">"\\.gz"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/basename.html">basename</a></span>(<span class="no">liftchain</span>)))
<span class="no">dfile</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span>(<span class="kw">fileext</span> <span class="kw">=</span> <span class="st">".gz"</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span>(<span class="no">liftchain</span>, <span class="no">dfile</span>)
<span class="kw pkg">R.utils</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/R.utils/man/compressFile.html">gunzip</a></span>(<span class="no">dfile</span>, <span class="kw">destname</span> <span class="kw">=</span> <span class="no">cloc38</span>, <span class="kw">remove</span> <span class="kw">=</span> <span class="fl">FALSE</span>)

<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">rtracklayer</span>)
<span class="no">chain38</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span>( <span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/Chain-class.html">import.chain</a></span>(<span class="no">cloc38</span>) )

<span class="co">## Run bulk data download (from S2B) to create gbm object</span>
<span class="kw">if</span> (!<span class="fu"><a href="https://rdrr.io/r/base/exists.html">exists</a></span>(<span class="st">"gbm"</span>)) <span class="no">gbm</span> <span class="kw">&lt;-</span> <span class="kw pkg">cBioPortalData</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/cBioPortalData/man/cBioDataPack.html">cBioDataPack</a></span>(<span class="st">"gbm_tcga"</span>)

<span class="no">mutations</span> <span class="kw">&lt;-</span> <span class="no">gbm</span><span class="kw">[[</span><span class="st">"mutations_extended"</span>]]
<span class="fu">seqlevelsStyle</span>(<span class="no">mutations</span>) <span class="kw">&lt;-</span> <span class="st">"UCSC"</span>

<span class="no">ranges38</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/liftOver.html">liftOver</a></span>(<span class="fu">rowRanges</span>(<span class="no">mutations</span>), <span class="no">chain38</span>)</pre></div>
</div>
</div>
<div id="figure-s3" class="section level2">
<h2 class="hasAnchor">
<a href="#figure-s3" class="anchor"></a>Figure S3</h2>
<div id="example-code-for-downloading-data-via-genomicdatacommons-and-loading-with-tcgautils" class="section level3">
<h3 class="hasAnchor">
<a href="#example-code-for-downloading-data-via-genomicdatacommons-and-loading-with-tcgautils" class="anchor"></a>Example code for downloading data via GenomicDataCommons and loading with TCGAutils</h3>
<div class="sourceCode" id="cb8"><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">TCGAutils</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">GenomicDataCommons</span>)

<span class="co">## GenomicDataCommons</span>
<span class="no">query</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/query.html">files</a></span>(<span class="kw">legacy</span> <span class="kw">=</span> <span class="fl">TRUE</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/filtering.html">filter</a></span>( ~ <span class="no">cases.project.project_id</span> <span class="kw">==</span> <span class="st">"TCGA-COAD"</span> <span class="kw">&amp;</span>
        <span class="no">data_category</span> <span class="kw">==</span> <span class="st">"Gene expression"</span> <span class="kw">&amp;</span>
        <span class="no">data_type</span> <span class="kw">==</span> <span class="st">"Exon quantification"</span> )

<span class="no">fileids</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/manifest.html">manifest</a></span>(<span class="no">query</span>)$<span class="no">id</span>[<span class="fl">1</span>:<span class="fl">4</span>]
<span class="no">exonfiles</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDataCommons/man/gdcdata.html">gdcdata</a></span>(<span class="no">fileids</span>, <span class="kw">use_cached</span> <span class="kw">=</span> <span class="fl">FALSE</span>)

<span class="co">## TCGAutils</span>
<span class="fu"><a href="https://rdrr.io/pkg/TCGAutils/man/makeGRangesListFromExonFiles.html">makeGRangesListFromExonFiles</a></span>(<span class="no">exonfiles</span>, <span class="kw">nrows</span> <span class="kw">=</span> <span class="fl">4</span>)
<span class="co">#&gt; Parsed with column specification:</span>
<span class="co">#&gt; cols(</span>
<span class="co">#&gt;   exon = col_character(),</span>
<span class="co">#&gt;   raw_counts = col_double(),</span>
<span class="co">#&gt;   median_length_normalized = col_double(),</span>
<span class="co">#&gt;   RPKM = col_double()</span>
<span class="co">#&gt; )</span>
<span class="co">#&gt; Parsed with column specification:</span>
<span class="co">#&gt; cols(</span>
<span class="co">#&gt;   exon = col_character(),</span>
<span class="co">#&gt;   raw_counts = col_double(),</span>
<span class="co">#&gt;   median_length_normalized = col_double(),</span>
<span class="co">#&gt;   RPKM = col_double()</span>
<span class="co">#&gt; )</span>
<span class="co">#&gt; Parsed with column specification:</span>
<span class="co">#&gt; cols(</span>
<span class="co">#&gt;   exon = col_character(),</span>
<span class="co">#&gt;   raw_counts = col_double(),</span>
<span class="co">#&gt;   median_length_normalized = col_double(),</span>
<span class="co">#&gt;   RPKM = col_double()</span>
<span class="co">#&gt; )</span>
<span class="co">#&gt; Parsed with column specification:</span>
<span class="co">#&gt; cols(</span>
<span class="co">#&gt;   exon = col_character(),</span>
<span class="co">#&gt;   raw_counts = col_double(),</span>
<span class="co">#&gt;   median_length_normalized = col_double(),</span>
<span class="co">#&gt;   RPKM = col_double()</span>
<span class="co">#&gt; )</span>
<span class="co">#&gt; GRangesList object of length 4:</span>
<span class="co">#&gt; $`TCGA-5M-AAT4-01A-11R-A41B-07`</span>
<span class="co">#&gt; GRanges object with 4 ranges and 3 metadata columns:</span>
<span class="co">#&gt;       seqnames      ranges strand | raw_counts median_length_normalized</span>
<span class="co">#&gt;          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; |  &lt;numeric&gt;                &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr1 11874-12227      + |          1                 0.135977</span>
<span class="co">#&gt;   [2]     chr1 12595-12721      + |          2                 0.547619</span>
<span class="co">#&gt;   [3]     chr1 12613-12721      + |          2                 0.472222</span>
<span class="co">#&gt;   [4]     chr1 12646-12697      + |          1                 0.529412</span>
<span class="co">#&gt;            RPKM</span>
<span class="co">#&gt;       &lt;numeric&gt;</span>
<span class="co">#&gt;   [1] 0.0228442</span>
<span class="co">#&gt;   [2] 0.1273517</span>
<span class="co">#&gt;   [3] 0.1483822</span>
<span class="co">#&gt;   [4] 0.1555160</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`TCGA-A6-6782-01A-11R-1839-07`</span>
<span class="co">#&gt; GRanges object with 4 ranges and 3 metadata columns:</span>
<span class="co">#&gt;       seqnames      ranges strand | raw_counts median_length_normalized</span>
<span class="co">#&gt;          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; |  &lt;numeric&gt;                &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr1 11874-12227      + |         35                 0.784702</span>
<span class="co">#&gt;   [2]     chr1 12595-12721      + |          9                 0.873016</span>
<span class="co">#&gt;   [3]     chr1 12613-12721      + |          9                 0.851852</span>
<span class="co">#&gt;   [4]     chr1 12646-12697      + |          8                 0.843137</span>
<span class="co">#&gt;            RPKM</span>
<span class="co">#&gt;       &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]  0.691243</span>
<span class="co">#&gt;   [2]  0.495456</span>
<span class="co">#&gt;   [3]  0.577274</span>
<span class="co">#&gt;   [4]  1.075605</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`TCGA-AA-3678-01A-01R-0905-07`</span>
<span class="co">#&gt; GRanges object with 4 ranges and 3 metadata columns:</span>
<span class="co">#&gt;       seqnames      ranges strand | raw_counts median_length_normalized</span>
<span class="co">#&gt;          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; |  &lt;numeric&gt;                &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr1 11874-12227      + |          4                 0.492918</span>
<span class="co">#&gt;   [2]     chr1 12595-12721      + |          2                 0.341270</span>
<span class="co">#&gt;   [3]     chr1 12613-12721      + |          2                 0.398148</span>
<span class="co">#&gt;   [4]     chr1 12646-12697      + |          2                 0.372549</span>
<span class="co">#&gt;            RPKM</span>
<span class="co">#&gt;       &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]  0.322477</span>
<span class="co">#&gt;   [2]  0.449436</span>
<span class="co">#&gt;   [3]  0.523655</span>
<span class="co">#&gt;   [4]  1.097661</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`TCGA-AA-3955-01A-02R-1022-07`</span>
<span class="co">#&gt; GRanges object with 4 ranges and 3 metadata columns:</span>
<span class="co">#&gt;       seqnames      ranges strand | raw_counts median_length_normalized</span>
<span class="co">#&gt;          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; |  &lt;numeric&gt;                &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr1 11874-12227      + |          0                        0</span>
<span class="co">#&gt;   [2]     chr1 12595-12721      + |          0                        0</span>
<span class="co">#&gt;   [3]     chr1 12613-12721      + |          0                        0</span>
<span class="co">#&gt;   [4]     chr1 12646-12697      + |          0                        0</span>
<span class="co">#&gt;            RPKM</span>
<span class="co">#&gt;       &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]         0</span>
<span class="co">#&gt;   [2]         0</span>
<span class="co">#&gt;   [3]         0</span>
<span class="co">#&gt;   [4]         0</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></pre></div>
</div>
</div>
<div id="figure-s4" class="section level2">
<h2 class="hasAnchor">
<a href="#figure-s4" class="anchor"></a>Figure S4</h2>
<div id="correlated-principal-components-across-experimental-assays-in-adrenocortical-carcinoma-acc" class="section level3">
<h3 class="hasAnchor">
<a href="#correlated-principal-components-across-experimental-assays-in-adrenocortical-carcinoma-acc" class="anchor"></a>Correlated principal components across experimental assays in adrenocortical carcinoma (ACC)</h3>
<p><a href="https://github.com/waldronlab/schematics/"><img src="https://raw.githubusercontent.com/waldronlab/schematics/master/pngs/unnamed-chunk-23-1.png"></a></p>
</div>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/LiNk-NY/curatedTCGAManu/">https://​github.com/​LiNk-NY/​curatedTCGAManu/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/LiNk-NY/curatedTCGAManu/issues">https://​github.com/​LiNk-NY/​curatedTCGAManu/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Marcel Ramos <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-3242-0582" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Levi Waldron <br><small class="roles"> Author </small>  </li>
<li><a href="authors.html">All authors...</a></li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://zenodo.org/badge/latestdoi/166464362"><img src="https://zenodo.org/badge/166464362.svg" alt="DOI"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by Marcel Ramos, Levi Waldron.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
