---
title: "eQTL PAN-CAN"
author: "Waldron Lab"
date: "March 6, 2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, echo = FALSE, warning = FALSE}
suppressPackageStartupMessages({
    library(MultiAssayExperiment)
    library(curatedTCGAData)
    library(TCGAutils)
    library(CNVRanger)
    library(BiocParallel)
})
```

Load packages:

```{r, eval = FALSE}
library(MultiAssayExperiment)
library(curatedTCGAData)
library(TCGAutils)
library(CNVRanger)
library(BiocParallel)
```

## Obtain ACC data from curatedTCGAData:
```{r}
data("diseaseCodes")
head(avail <- diseaseCodes[
    diseaseCodes[["Available"]] == "Yes", "Study.Abbreviation"])
## Peaks data not available in LAML or SKCM
avail <- avail[!avail %in% c("LAML", "SKCM")]
```

```{r}
.addCallStates <-
    function(mae, calls_assay, sample = "01", col = "Segment_Mean") {
    calls_assay <- paste0(sample, "_", calls_assay)
    state <- round(mcols(mae[[calls_assay]])[[col]])
    state[state > 2] <- 2
    state[state < -2] <- -2
    mcols(mae[[calls_assay]])$State <- state + 2
    mcols(mae[[calls_assay]]) <-
        mcols(mae[[calls_assay]])[
            c("State", "Segment_Mean", "Num_Probes")]
    mae
}

.fixRowRanges <- function(mae, cnvrs) {
    peakObj <- mae[[cnvrs]]
    if (is.null(rowRanges(peakObj))) {
        peakrow <- rowData(peakObj)
        rows <- peakrow[,
            grepl("gene|ranges", names(peakrow), ignore.case = TRUE)]
        if (length(rows)) {
            rowRanges(peakObj) <- as(rows, "GRanges")
            rowData(peakObj) <- peakrow
        }
    }
    mae[[cnvrs]] <- peakObj
    mae
}
```

MultiOmic Quality Control

```{r}
# BiocManager::install("Bioconductor/MultiOmicQC")
library(MultiOmicQC)
```

eQTL Analysis


```{r}
eQTL_pan <- function(ccode, sample = "01") {
    mae <- curatedTCGAData::curatedTCGAData(ccode,
        assays = c("GISTIC_Peaks", "CNVSNP", "RNASeq2GeneNorm"),
        dry.run = FALSE)
    scalls <- paste0(sample, "_", ccode, "_CNVSNP-20160128")
    rrnas <- paste0(sample, "_", ccode, "_RNASeq2GeneNorm-20160128",
        "_ranged")
    scnvrs <- paste0(sample, "_", ccode, "_GISTIC_Peaks-20160128")

    mae <- symbolsToRanges(mae, unmapped = FALSE)
    genome(mae) <- "hg19"
    seqlevelsStyle(mae) <- "UCSC"

    mae <- intersectColumns(mae)
    mae <- splitAssays(mae, sampleCodes=sample)
    mae <- .addCallStates(mae, paste0(ccode, "_CNVSNP-20160128"))
    mae <- .fixRowRanges(mae, scnvrs)
    CNVRanger::cnvExprAssoc(
        cnvrs = scnvrs,
        calls = scalls,
        rcounts = rrnas,
        data = mae
    )
}
```

```{r}
param <- MulticoreParam(workers = 14, stop.on.error = FALSE)
pancan <- bptry(
    bplapply(avail, eQTL_pan, BPPARAM = param)
)

save(pancan, file = "pancanres.rda")
```

```{r}
pancan.redo <- bplapply(avail, eQTL_pan, BPREDO = pancan, BPPARAM = param)
save(pancan.redo, file = "pancan_second_try.rda")
```
