---
title: "Example Use Case to counter GMQL"
author: "Waldron Lab"
date: "July 11, 2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Load packages:

```{r,results="hide",include=TRUE,message=FALSE,warning=FALSE}
library(MultiAssayExperiment)
library(curatedTCGAData)
library(TCGAutils)
```

From [Masseroli et al. 2018](https://doi.org/10.1093/bioinformatics/bty688)

> "In TCGA data of BRCA patients, find the DNA somatic mutations
 within the first 2000 bp outside of the genes that are both
 expressed with FPKM > 3 and have at least a methylation in the same patient
 biospecimen, and extract these mutations of the top 5% patients
 with the highest number of such mutations."

First we load the 'BRCA' dataset with 'Mutation', 'RNASeq2GeneNorm' and
'Methylation' assays / experiments:

```{r}
system.time(
    brca <- curatedTCGAData(
        diseaseCode = "BRCA",
        assays = c("Mutation", "RNASeq2GeneNorm", "Methylation"),
        dry.run = FALSE
    )
)
```

We then add gene symbol annotations with genomic ranges and drop any unmapped
symbols. We also do the same for CpG probe identifiers:

```{r}
brca0 <- symbolsToRanges(brca, unmapped = FALSE)

if (!require("IlluminaHumanMethylation450kanno.ilmn12.hg19", quietly = TRUE))
    BiocManager::install("IlluminaHumanMethylation450kanno.ilmn12.hg19")

library(IlluminaHumanMethylation450kanno.ilmn12.hg19)

brca1 <- CpGtoRanges(brca0, unmapped = FALSE)
```

Select only tumors in the data

```{r}
library(TCGAutils)
sampleTables(brca1)
head(sampleTypes)

sampleselector <- lapply(colnames(brca1), function(x)
    TCGAsampleSelect(x, c("01", "06"))
)
brca1 <- brca1[, sampleselector]
```

We check what probes "have at least a methylation..." across all samples:

```{r}
unmeth <- is.na(rowSums(
    assay(brca1[["BRCA_Methylation_methyl450-20160128_ranged"]])
))
methyl <- brca1[["BRCA_Methylation_methyl450-20160128_ranged"]][!unmeth, ]
brca1[["BRCA_Methylation_methyl450-20160128_ranged"]] <- methyl
brca1[["BRCA_Methylation_methyl450-20160128_ranged"]]
```

An expression threshold is used where "FPKM > 3". In our case, we use the better
alternative: TPM > 3

https://haroldpimentel.wordpress.com/2014/05/08/what-the-fpkm-a-review-rna-seq-expression-units/
http://diytranscriptomics.com/Reading/files/wagnerTPM.pdf

```{r}
validRows <- apply(
    assay(brca1[["BRCA_RNASeq2GeneNorm-20160128_ranged"]]) > 3, 1L, any)
brca1[["BRCA_RNASeq2GeneNorm-20160128_ranged"]] <-
    brca1[["BRCA_RNASeq2GeneNorm-20160128_ranged"]][validRows, ]
brca1[["BRCA_RNASeq2GeneNorm-20160128_ranged"]]
```

Restrict Methylation to genes with expression values > 3

```{r}
rnaranges <- rowRanges(brca1[["BRCA_RNASeq2GeneNorm-20160128_ranged"]])
methranges <- rowRanges(brca1[["BRCA_Methylation_methyl450-20160128_ranged"]])
hits <- findOverlaps(rnaranges, methranges)
methdx <- sort(unique(subjectHits(hits)))

methsub <- brca1[["BRCA_Methylation_methyl450-20160128_ranged"]][methdx, ]
brca1[["BRCA_Methylation_methyl450-20160128_ranged"]] <- methsub
brca1
```

Find matching genes in mutations from RNA

```{r}
validMuts <-
    mcols(brca1[["BRCA_Mutation-20160128"]])[["Hugo_Symbol"]] %in%
        names(rnaranges)

muts <- brca1[["BRCA_Mutation-20160128"]][validMuts, ]
mutsdown <- flank(rowRanges(muts), 2000, start = FALSE)
mutsup <- flank(rowRanges(muts), 2000)
allROIs <- c(mutsdown, mutsup)
updownmuts <- muts %within% allROIs
brcamuts <- muts[updownmuts, ]

reps <- replicated(brca1)[["BRCA_Mutation-20160128"]]
mutbyregion <- assay(brcamuts, i = "Variant_Classification")
mutTally <- ifelse(!is.na(mutbyregion), 1, 0)
rseMut <- SummarizedExperiment(assay = mutTally,
    rowRanges = as(rownames(mutTally), "GRanges"))

# Find 5 percent of patients with most mutations
compReps <- mergeReplicates(mutTally, reps, sum)
mutsPerPt <- rowSums(t(compReps))

top5 <- head(sort(mutsPerPt, decreasing = TRUE), ceiling(length(mutsPerPt)*0.05))

pats <- TCGAbarcode(names(top5))

brca2 <- c(brca1, flankMutations = rseMut, mapFrom = 1L)
brcaf <- brca2[, rownames(colData(brca2)) %in% pats, ]
rowRanges(brcaf[["flankMutations"]])
```


