---
title: "Figures 1 to 4"
author: "Waldron Lab"
date: "July 23, 2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
Load packages:

```{r,results="hide",include=TRUE,message=FALSE,warning=FALSE}
library(MultiAssayExperiment)
library(curatedTCGAData)
library(TCGAutils)
library(survival)
```
# Using ACC

```{r,results="hide",include=TRUE,message=FALSE,warning=FALSE}
acc <- curatedTCGAData("ACC", c("RNASeq2GeneNorm", "GISTIC_T*"), FALSE)
```

## Figure 1 Example: Multivariate Cox regression with RNASeq expression, copy number, and pathology

```{r}
wideacc <- wideFormat(acc["EZH2", , ],
    colDataCols=c("vital_status", "days_to_death", "pathology_N_stage"))
wideacc[["y"]] <- Surv(wideacc$days_to_death, wideacc$vital_status)
wideacc
coxph(
    Surv(days_to_death, vital_status) ~ `ACC_GISTIC_ThresholdedByGene.20160128_EZH2` +
        log2(`ACC_RNASeq2GeneNorm.20160128_EZH2`) + pathology_N_stage,
    data=wideacc
)
```

## Figure 2 Example: Correlation between RNASeq expression and copy number

```{r}
subacc <- intersectColumns(acc)
subacc <- intersectRows(subacc)
```

Create a list of numeric matrices:

```{r}
acc.mats <- assays(subacc)
```


Log-transform the RNA-seq assay:
```{r}
acc.mats[["ACC_RNASeq2GeneNorm-20160128"]] <-
    log2(acc.mats[["ACC_RNASeq2GeneNorm-20160128"]] + 1)
```

Transpose both, so genes are in columns:

```{r}
subacc.list <- lapply(acc.mats, t)
```

Calculate the correlation between columns in the first matrix and columns in the
second matrix:


```{r}
corres <- cor(subacc.list[[1]], subacc.list[[2]])
```

And finally, create the histograms:

```{r}
library(ggplot2)
diagvals <- diag(corres)
diagframe <- data.frame(genename = names(diagvals), value = diagvals)

png("rnaseq-cn-correl.png", width = 8, height = 6, units = "in", res = 300)
ggplot(diagframe, aes(x = value)) + geom_histogram(color = "black",
    fill = "white", binwidth = 0.082) + theme_bw() +
    ggtitle("Correlation between RNASeq and Copy Number Values in ACC") +
    xlab(expression(paste("Pearson coefficient (", rho, ")"))) +
    ylab("Frequency")

dev.off()
```

```{r}
hist(corres[upper.tri(corres)])
```

## Figure 3 Example: Violin plot of log2 expression by copy number

Gene with the highest correlation to CN, make boxplot of log2 expression vs
copy number

```{r}
(topvals <- head(sort(abs(diag(corres)), decreasing = TRUE)))
```

The 'SNRPB2' gene expression has a correlation to CNV = 0.83

```{r}
snr <- subacc[names(topvals)[1L], ]
```

```{r}
snpexp <- as.data.frame(wideFormat(snr, check.names = FALSE))
names(snpexp) <- c("primary", "CopyNumber", "Expression")
## remove outlier in CN
table(snpexp$CopyNumber)
snpexp <- snpexp[snpexp$CopyNumber != -2, ]

snpexp[["Log2Exp"]] <- log(snpexp[["Expression"]], 2)
library(ggplot2)
png("violin_exp.png")
ggplot(snpexp, aes(factor(CopyNumber), Log2Exp)) +
    geom_violin() +
    theme_classic() +
    ggtitle("Log2 Expression values by Copy Number for SNRPB2") +
    xlab("Copy Number") +
    ylab("log2(expression)") +
    geom_boxplot(width=0.1)
dev.off()
```

## Figure 4 example: Identifying correlated principal components

Perform Principal Components Analysis of each of the five assays, using samples
available on each assay, log-transforming RNA-seq data first.  Using the first
10 components, calculate Pearson correlation between all scores and plot these
correlations as a heatmap to identify correlated components across assays.

```{r}
acc_full <- curatedTCGAData("ACC",
    assays = c("RNASeq2*", "GISTIC_T*", "RPPA*", "Mutation", "miRNASeqGene"),
    dry.run = FALSE)
```

```{r}
getLoadings <- function(x, ncomp=10, dolog=FALSE, center=TRUE, scale.=TRUE) {
    if (dolog){
        x <- log2(x + 1)
    }
    pc <- prcomp(x, center=center, scale.=scale.)
    t(pc$rotation[, 1:10])
}
```

```{r}
acc2 <- intersectColumns(acc_full)
```

Categorize by silent mutations:

```{r}
mut <- assay(acc2[["ACC_Mutation-20160128"]], i = "Variant_Classification")
mut <- ifelse(is.na(mut) | mut == "Silent", 0, 1)
acc2[["ACC_Mutation-20160128"]] <- mut
```

```{r}
acc2 <- c(acc2,
    `RNA-Seq` = getLoadings(assays(acc2)[["ACC_RNASeq2GeneNorm-20160128"]],
        dolog=TRUE), mapFrom=1L
    )
acc2 <- c(acc2,
    `GISTIC-T` = getLoadings(assays(acc2)[["ACC_GISTIC_ThresholdedByGene-20160128"]],
        center=FALSE, scale.=FALSE), mapFrom=2L)
acc2 <- c(acc2,
    Protein = getLoadings(assays(acc2)[["ACC_RPPAArray-20160128"]]),
        mapFrom=3L)
acc2 <- c(acc2,
    Mutations = getLoadings(assays(acc2)[["ACC_Mutation-20160128"]],
        center=FALSE, scale.=FALSE), mapFrom=4L)
acc2 <- c(acc2,
    miRNA = getLoadings(assays(acc2)[["ACC_miRNASeqGene-20160128"]]),
        mapFrom=5L)
```

Now subset to keep *only* the PCA results:
```{r}
acc2 <- acc2[, , 6:10]
experiments(acc2)
```

Note, it would be equally easy (and maybe better) to do PCA on all samples
available for each assay, then do intersectColumns at this point instead.

Now, steps for calculating the correlations and plotting a heatmap:
* Use *wideFormat* to paste these together, which has the nice property of
adding assay names to the column names.
* The first column always contains the sample identifier, so remove it.
* Coerce to a matrix
* Calculate the correlations, and take the absolute value (since signs of
principal components are arbitrary)
* Set the diagonals to NA (each variable has a correlation of 1 to itself).

```{r}
df <- wideFormat(acc2)[, -1]
mycors <- cor(as.matrix(df))
mycors <- abs(mycors)
diag(mycors) <- NA
```

To simplify the heatmap, show only components that have at least one correlation
greater than 0.5.
```{r}
has.high.cor <- apply(mycors, 2, max, na.rm=TRUE) > 0.5
mycors <- mycors[has.high.cor, has.high.cor]

# colors
library(RColorBrewer)
mycolors <- colorRampPalette(brewer.pal(n = 7, name = "Oranges"))(100)

png("pca_correls.png", width = 9, height = 8, units = "in", res = 300)
pheatmap::pheatmap(mycors, color = mycolors,
    main = "Correlated principal components across experimental assays",
    angle_col = 90)
dev.off()
```
