---
title: "OncoPrint Figure"
author: "Waldron Lab"
date: "July 23, 2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
Load packages:

```{r,results="hide",include=TRUE,message=FALSE,warning=FALSE}
library(MultiAssayExperiment)
library(curatedTCGAData)
library(TCGAutils)
library(ComplexHeatmap)
library(RColorBrewer)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
library(org.Hs.eg.db)
library(GenomeInfoDb)
```

# Using ACC

```{r,results="hide",include=TRUE,message=FALSE,warning=FALSE}
acc <- curatedTCGAData("ACC",
    c("RNASeq2GeneNorm", "GISTIC_T*", "Mutation"), FALSE)
```

```{r}
## remove single Translation_Site_* mutation
TranslationSite <-
    mcols(acc[["ACC_Mutation-20160128"]])$Variant_Classification ==
        "Translation_Start_Site"
mcols(acc[["ACC_Mutation-20160128"]])$Variant_Classification[TranslationSite] <-
    NA_character_

variants <- mcols(acc[["ACC_Mutation-20160128"]])$Variant_Classification
(mutclass <- table(variants))

length(mutclass)
vclass <- setNames(names(mutclass), names(mutclass))
```

```{r}
gn <- genes(TxDb.Hsapiens.UCSC.hg19.knownGene)
gn <- keepStandardChromosomes(granges(gn), pruning.mode="coarse")
seqlevelsStyle(gn) <- "NCBI"
names(gn) <- AnnotationDbi::mapIds(org.Hs.eg.db::org.Hs.eg.db, names(gn),
    keytype = "ENTREZID", column = "SYMBOL")
gn <- sort(gn)

simplify_funs <- lapply(vclass, function(variant) {
    args <- alist(scores =, ranges =, qranges =)
    args <- as.pairlist(args)
    body <- substitute({
       any(scores %in% z)
    }, list(z = variant))
    eval(call("function", args, body))
})

ragex <- acc[["ACC_Mutation-20160128"]]
genome(ragex) <- translateBuild(genome(ragex))

list_mats <- lapply(simplify_funs, function(variant_fun) {
    res <- qreduceAssay(ragex, gn, variant_fun, "Variant_Classification")
    rownames(res) <- names(gn)
    res[is.na(res)] <- 0
    res[!is.na(rownames(res)), ]
})

nomutgenes <- Reduce(intersect, lapply(list_mats, function(x) rownames(x[rowSums(x) == 0, ])))

updated_mats <- lapply(list_mats, function(genemat) {
    genemat[!rownames(genemat) %in% nomutgenes, ]
})
```

```{r}
qualcolors <- brewer.pal(n = 9, "Set3")
colors <- setNames(qualcolors, names(mutclass))
```

```{r}
mutfuns <- lapply(colors, function(couleur) {
    args <- alist(x =, y =, w =, h =)
    args <- as.pairlist(args)
    body <- substitute({
        grid::grid.rect(x, y, w, h, gp = gpar(fill = z, col = NA))
    }, list(z = couleur))
    eval(call("function", args, body))
})
```

```{r}
tops <- ceiling(nrow(updated_mats[[1]]) * 0.0025)
mgenes <- Reduce(union, lapply(updated_mats, function(x)
    names(sort(rowSums(x), decreasing = TRUE)[seq_len(tops)]))
)
length(mgenes)
upmats2 <- lapply(updated_mats, function(mat) mat[mgenes, ])
```

```{r}
oncoPrint(upmats2, alter_fun = mutfuns, col = colors)
```

